src/cargo/util/toml/mod.rs

```diff
@@ -867,8 +868,8 @@ impl TomlDependency {
             }
         }
 
-        let new_source_id = match (details.git.as_ref(), details.path.as_ref()) {
-            (Some(git), maybe_path) => {
+        let new_source_id = match (details.git.as_ref(), details.path.as_ref(), details.registry.as_ref()) {
+            (Some(git), maybe_path, _) => {
```

Comment:
I think we'll want to match this here as well, and without backwards compatiblity concerns (like the `path` dependency) we should just return an error for both `git = "foo", registry = "bar"` in a dependency specification

Reply:
Yes I think this code is also the source of the bad error message @carols10cents mentioned above.